source/extensions/filters/network/zookeeper_proxy/zookeeper_decoder.h

```diff
@@ -0,0 +1,128 @@
+#pragma once
+#include <cstdint>
+
+#include "envoy/common/platform.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/enum_to_int.h"
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+enum class XidCodes {
+  CONNECT_XID = 0,
+  WATCH_XID = -1,
+  PING_XID = -2,
+  AUTH_XID = -4,
+  SET_WATCHES_XID = -8
+};
+
+enum class OpCodes {
+  CONNECT = 0,
+  CREATE = 1,
+  DELETE = 2,
+  EXISTS = 3,
+  GETDATA = 4,
+  SETDATA = 5,
+  GETACL = 6,
+  SETACL = 7,
+  GETCHILDREN = 8,
+  SYNC = 9,
+  PING = 11,
+  GETCHILDREN2 = 12,
+  CHECK = 13,
+  MULTI = 14,
+  CREATE2 = 15,
+  RECONFIG = 16,
+  CHECKWATCHES = 17,
+  REMOVEWATCHES = 18,
+  CLOSE = -11,
+  SETAUTH = 100,
+  SETWATCHES = 101
+};
+
+enum class WatcherType { CHILDREN = 1, DATA = 2, ANY = 3 };
+
+/**
+ * General callbacks for dispatching decoded ZooKeeper messages to a sink.
+ */
+class DecoderCallbacks {
+public:
+  virtual ~DecoderCallbacks() {}
+
+  virtual void onDecodeError() PURE;
+  virtual void onRequestBytes(const uint64_t bytes) PURE;
```

Comment:
We tend to avoid using const values in signatures (see https://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html)